### PR TITLE
Issue-2805: Simplify JUnit `assertTrue/assertFalse`

### LINKED ISSUE #2805: Simplify JUnit `assertTrue/assertFalse` to the corresponding dedicated assertion
We have these assertions in `ConfigMapTest`:
https://github.com/fabric8io/kubernetes-client/blob/b015fb957047956d3656c3043e4c663957e82de9/kubernetes-model-generator/kubernetes-model-core/src/test/java/io/fabric8/kubernetes/api/model/ConfigMapTest.java#L71-L74

Instead of `assertTrue()` we should use `assertEquals` for comparing while asserting expected results. You can read more about this here[0]. There are other places too where this pattern is followed[1], we should refactor these to use dedicated exception.

[0] https://sonarcloud.io/organizations/fabric8io/rules?open=java%3AS5785&rule_key=java%3AS5785
[1] https://sonarcloud.io/organizations/fabric8io/issues?projects=fabric8io_kubernetes-client&resolved=false&rules=java%3AS5785

### PR DESCRIPTION
**Pull request by amitdaga**
Fix #2805 

## Description

This PR simplifies few `assertTrue` and `assertFalse` assertions to instead use `assertEquals` and `assertNotEquals` methods respectively. More details are present in the linked issue.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by amitdaga** [comment]
Imports were organized in all the touched classes in this PR. Hope that's fine.

**Review by oscerd — APPROVED** [review]

**Comment by amitdaga** [comment]
The test failure for `E2E Tests / K8S (v1.12.0)` seems to be unrelated. Please let me know if you would still like me to investigate this failure.

File: kubernetes-model-generator/kubernetes-model-core/src/test/java/io/fabric8/kubernetes/api/model/QuantityTest.java [hunk 1]
```diff
@@ -117,13 +114,13 @@ public void testExponents() {
   @Test
   @DisplayName("Test equals operation")
   public void testEquality() {
-    assertTrue(new Quantity(".5Mi").equals( new Quantity("512Ki")));
-    assertTrue(new Quantity("1Gi").equals(new Quantity("1024Mi")));
-    assertFalse(new Quantity("1E").equals(null));
-    assertFalse(((Object)new Quantity("2P")).equals((Object)"2P"));
+    assertEquals(new Quantity(".5Mi"), new Quantity("512Ki"));
+    assertEquals(new Quantity("1Gi"), new Quantity("1024Mi"));
+    assertNotEquals(null, new Quantity("1E"));
```
Comment (manusa) [inline kubernetes-model-generator/kubernetes-model-core/src/test/java/io/fabric8/kubernetes/api/model/QuantityTest.java:119]:
```suggestion
    assertNotNull(new Quantity("1E"));
```

Comment (amitdaga) [reply kubernetes-model-generator/kubernetes-model-core/src/test/java/io/fabric8/kubernetes/api/model/QuantityTest.java:119]: @manusa in this line, an object is newly created, and the code is trying to assert whether it's not null. The assertion seems to be redundant and can be removed itself. Let me know if that seems okay.

Comment (manusa) [reply kubernetes-model-generator/kubernetes-model-core/src/test/java/io/fabric8/kubernetes/api/model/QuantityTest.java:119]: Yes, that's exactly the way to go.

Comment (amitdaga) [reply kubernetes-model-generator/kubernetes-model-core/src/test/java/io/fabric8/kubernetes/api/model/QuantityTest.java]: This is done.

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**